ZetaPushSwift/ZetaPushMacroService.swift

```diff
@@ -72,307 +84,265 @@ public enum ZetaPushMacroError: Error {
+        
+        guard let subscription = sub else { return }
+        self?.clientHelper?.unsubscribe(subscription)
+        
+        if let result = messageDict["result"] as? NSDictionary {
+          seal.fulfill(result)
         }
-        if subscriptionArray?.count == 0 {
-            self.channelSubscriptionBlocks[subscription.channel] = nil;
+        guard let errors = messageDict["errors"] as? [[String: Any]], errors.isEmpty else {
+          return
         }
-    }
-    
-    open func call(verb:String, parameters:NSDictionary) {
-        let dict:[String:AnyObject] = [
-            "name": verb as AnyObject,
-            "hardFail": true as AnyObject,
-            "parameters": parameters
-        ]
-        self.clientHelper?.publish(composeServiceChannel("call"), message: dict)
-    }
-    
-    /*
-        Call return a promise
-     */
-    open func call(verb:String, parameters:[String:AnyObject]) -> Promise<NSDictionary> {
-        return Promise { seal in
-            let requestId = UUID().uuidString
-            
-            let dict:[String:AnyObject] = [
-                "name": verb as AnyObject,
-                "hardFail": false as AnyObject,
-                "parameters": parameters as AnyObject,
-                "requestId": requestId as AnyObject
-            ]
-            
-            var sub: Subscription? = nil
-            
-            let channelBlockMacroCall:ChannelSubscriptionBlock = {(messageDict) -> Void in
-                
-                // Check if the requestId is similar to the one sent
-                if messageDict.object(forKey: "requestId") != nil {
-                    if let msgRequestId = messageDict["requestId"] as? String {
-                        if msgRequestId != requestId {
-                            return
-                        }
-                    }
-                }
-                
-                self.clientHelper?.unsubscribe(sub!)
-                
-                if let result = messageDict["result"] as? NSDictionary {
-                    seal.fulfill(result)
-                }
-                if messageDict.object(forKey: "errors") != nil {
-                    if let errors = messageDict["errors"] as? NSArray {
-                        if errors.count > 0 {
-                             if let error = errors[0] as? NSDictionary {
-                                seal.reject(ZetaPushMacroError.genericFromDictionnary(error))
-                             } else {
-                                seal.reject(ZetaPushMacroError.unknowError)
-                             }
-                        }
-                    }
-                }
-                
-            }
-            
-            sub = self.clientHelper?.subscribe(composeServiceChannel(verb), block: channelBlockMacroCall)
- 
-            self.clientHelper?.publish(composeServiceChannel("call"), message: dict)
-            
+        if let error = errors.first as? NSDictionary {
```

Comment:
for this you can add previous guard let to do the same in a simplte if let. This modification is better to manage unknown error.